fe/fe-core/src/main/java/com/starrocks/task/ExportExportingTask.java

```diff
@@ -338,19 +340,31 @@ protected void exec() {
                 }
 
                 if (i < RETRY_NUM - 1) {
-                    TUniqueId queryId = coord.getQueryId();
-                    coord.clearExportStatus();
-
-                    // generate one new queryId here, to avoid being rejected by BE,
-                    // because the request is considered as a repeat request.
-                    // we make the high part of query id unchanged to facilitate tracing problem by log.
+                    TUniqueId oldQueryId = coord.getQueryId();
                     UUID uuid = UUID.randomUUID();
-                    TUniqueId newQueryId = new TUniqueId(queryId.hi, uuid.getLeastSignificantBits());
-                    coord.setQueryId(newQueryId);
+                    TUniqueId newQueryId = new TUniqueId(oldQueryId.hi, uuid.getLeastSignificantBits());
+                    if (exportJob.needResetCoord()) {
+                        try {
+                            Coordinator newCoord = exportJob.resetCoord(taskIdx, newQueryId);
+                            coord = newCoord;
+                        } catch (UserException e) {
+                            LOG.warn("fail to reset coord for task idx: {}, task query id: {}, reason: {}", taskIdx, 
+                                    getQueryId(), e.getMessage());
+                            // still use old coord if there are any problems when reseting Coord
+                            coord.clearExportStatus();
+                            coord.setQueryId(newQueryId);
+                        }
+                    } else {
+                        coord.clearExportStatus();
+                        // generate one new queryId here, to avoid being rejected by BE,
+                        // because the request is considered as a repeat request.
+                        // we make the high part of query id unchanged to facilitate tracing problem by log.
+                        coord.setQueryId(newQueryId);
+                    }
```

Comment:
```suggestion
                    if (exportJob.needResetCoord()) {
                        try {
                            Coordinator newCoord = exportJob.resetCoord(taskIdx, newQueryId);
                            coord = newCoord;
                        } catch (UserException e) {
                            LOG.warn("fail to reset coord for task idx: {}, task query id: {}, reason: {}", taskIdx, 
                                    getQueryId(), e.getMessage());
                            // still use old coord if there are any problems when reseting Coord
                            coord.clearExportStatus();
                        }
                    } else {
                        coord.clearExportStatus();
                        // generate one new queryId here, to avoid being rejected by BE,
                        // because the request is considered as a repeat request.
                        // we make the high part of query id unchanged to facilitate tracing problem by log.
                    }
                    coord.setQueryId(newQueryId);
```